sdk/data/azcosmos/cosmos_container_properties_test.go

```diff
@@ -66,92 +66,41 @@ func TestContainerPropertiesSerialization(t *testing.T) {
-	}
-
-	if otherProperties.ConflictResolutionPolicy.ResolutionPath != properties.ConflictResolutionPolicy.ResolutionPath {
-		t.Errorf("Expected ConflictResolutionPolicy.ResolutionPath to be %s, but got %s", properties.ConflictResolutionPolicy.ResolutionPath, otherProperties.ConflictResolutionPolicy.ResolutionPath)
+	tests := []struct {
+		name     string
+		got      interface{}
+		expected interface{}
+	}{
+		{"ID", otherProperties.ID, properties.ID},
+		{"ETag", *otherProperties.ETag, *properties.ETag},
+		{"SelfLink", otherProperties.SelfLink, properties.SelfLink},
+		{"ResourceID", otherProperties.ResourceID, properties.ResourceID},
+		{"LastModified", otherProperties.LastModified, properties.LastModified},
+		{"AnalyticalStoreTimtoLiveInSeconds", otherProperties.AnalyticalStoreTimeToLiveInSeconds, properties.AnalyticalStoreTimeToLiveInSeconds},
+		{"DefaultTimeToLive", otherProperties.DefaultTimeToLive, properties.DefaultTimeToLive},
+		{"PartitionKeyDefinitionPaths", otherProperties.PartitionKeyDefinition.Paths[0], properties.PartitionKeyDefinition.Paths[0]},
+		{"PartitionKeyDefinitionVersion", otherProperties.PartitionKeyDefinition.Version, properties.PartitionKeyDefinition.Version},
+		{"IndexingPolicy", otherProperties.IndexingPolicy != nil, properties.IndexingPolicy != nil},
+		{"IndexingPolicyAutomatic", otherProperties.IndexingPolicy.Automatic, properties.IndexingPolicy.Automatic},
+		{"IndexingPolicyIndexingMode", otherProperties.IndexingPolicy.IndexingMode, properties.IndexingPolicy.IndexingMode},
+		{"IndexingPolicyIncludedPaths", otherProperties.IndexingPolicy.IncludedPaths[0], properties.IndexingPolicy.IncludedPaths[0]},
+		{"IndexingPolicyExcludedPaths", otherProperties.IndexingPolicy.ExcludedPaths[0], properties.IndexingPolicy.ExcludedPaths[0]},
+		{"IndexingPolicySpatialIndexesPath", otherProperties.IndexingPolicy.SpatialIndexes[0].Path, properties.IndexingPolicy.SpatialIndexes[0].Path},
+		{"IndexingPolicySpatialIndexesSpatialTypes", otherProperties.IndexingPolicy.SpatialIndexes[0].SpatialTypes[0], properties.IndexingPolicy.SpatialIndexes[0].SpatialTypes[0]},
+		{"IndexingPolicyCompositeIndexesPath", otherProperties.IndexingPolicy.CompositeIndexes[0][0].Path, properties.IndexingPolicy.CompositeIndexes[0][0].Path},
+		{"UniqueKeyPolicy", otherProperties.UniqueKeyPolicy != nil, properties.UniqueKeyPolicy != nil},
+		{"UniqueKeyPolicyUniqueKeys", otherProperties.UniqueKeyPolicy.UniqueKeys[0].Paths[0], properties.UniqueKeyPolicy.UniqueKeys[0].Paths[0]},
+		{"ConflictResolutionPolicy", otherProperties.ConflictResolutionPolicy != nil, properties.ConflictResolutionPolicy !=nil},
+		{"ConflictResolutionPolicyMode", otherProperties.ConflictResolutionPolicy.Mode, properties.ConflictResolutionPolicy.Mode},
+		{"ConflictResolutionPolicyResolutionPath", otherProperties.ConflictResolutionPolicy.ResolutionPath, properties.ConflictResolutionPolicy.ResolutionPath},
+	}
+	
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
I personally find this refactoring a bit less clear than the `if`s. Having to understand the usage of `t.Run` with the delegate seems a bit more complicated than understanding a list of `if` conditions.

But I would see value if these `if`s are unbounded/can keep growing, so it would reduce the volume of code required. But the reality is that these properties will not really change frequently so the chances of a developer having to keep adding checks here is very low to offset the increase in reading complexity.

Finally, the same pattern (`if`s) are used throughout the other files. If we want to think a common pattern, I guess we would need to refactor all the other files too?